functional-programming-spreadsheet/index860.js

```diff
@@ -0,0 +1,111 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const isEven = num => num % 2 === 0;
+
+const spreadsheetFunctions = {
+  "": x => x,
+  random: ([x, y]) => Math.floor(Math.random() * y + x),
+  increment: nums => nums.map(x => x + 1),
+  firsttwo: arr => arr.slice(0, 2),
+  lastttwo: arr => arr.slice(-2)
+};
+
+/*
+The `filter` method keeps only the elements of an array that satisfy the function passed to it:
+```
+[1, 10, 8, 3, 4, 5].filter(x > 3) = [10, 8, 4, 5]
+```
+Use `filter` to add a function called `even` to `spreadsheetFunctions`, which returns all the even elements of an array.
```

Comment:
Use `filter` to add a function called `even` to `spreadsheetFunctions`, which returns all the even elements of an array, `nums`.